### ISSUE TITLE
Implement `Clone` for `CallBuilder`

### ISSUE THREAD

**Issue by sampersand** [SergejJurecko/mio_httpc#6]
This trait, along with a myriad of other `#[derive]`-able ones are missing from some of the core structs.

There's no reason not to do so?

**Comment by SergejJurecko** [comment]
Clone can easily be added to CallBuilder. What else makes sense? Just adding them because they can be added only results in longer compile times and a larger binary potentially.

**Comment by sampersand** [comment]
Take a look at https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-common-traits

tl;dr users of your crate are unable to implement traits defined in other crates for your types (eg `Clone`)

You never know what traits your end users might need, so `#[derive]` all of them that make sense (or `impl` if need be)

For example, I needed `Clone` because I was sending POST requests to the same server with the same timeout and everything; the only difference is what the payload was. Having to create (ie not `clone()`) a new `CallBuilder` every time was too inefficient

The compile time overhead shouldn't be that big of an issue. For executable size, if the trait isn't being used in the final binary (don't quote me on this part) I'm pretty sure it isn't compiled in. But even if it is, it wouldn't be that much of an issue (and the benefits would outweigh the negatives)

**Comment by SergejJurecko** [comment]
Released 0.6.16 with updated rustls and clone on builder
